module-4/backend/week_one.md

```diff
@@ -0,0 +1,50 @@
+## Week One - Module 4 Recap
+
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+
+Note: When you're done, submit a PR. 
+
+1. What's the most useful thing you learned from completing the intermission week work?
+- basic javascript - for loops, forEach, functions, difference between objects in js and hashes in ruby
+
+2. What are some tools to help debug JavaScript code?
+- pry.js, console.log
+
+3. What are some tools you need in order to unit test your JavaScript?
+- mocha and chai 
+
+4. What is the syntax for invoking a function?
+functionName()
+
+5. What's the difference between `==` and `===` in JavaScript?
+- == checks if something is equal to something else, not strict. Example: 2 == 2 //true or true == 1 //true
+- === checks if something is equal to something else, is strict, meaning it also checks the datatypes Example: true === 1 //false because true is a boolean and 1 is an integer
+
+6. What's the difference between asynchronous and synchronous JavaScript? 
+- asynchronous - does not wait for a function to finish before moving on to the next. 
+- synchronous - does wait for a function to finish before moving on. 
+
+7. How do we setup a route when creating an API with Node and Express?
+- in the server.js file set var app = Express() then you can set app.get or app.post and pass in the url and tell it what to response with (response code and/or some data via a query to the database) 
```

Comment:
Remember that `request` and `response` are required variables for the callback function.